metrics_utility/test/test_extra_params_validation.py

```diff
@@ -0,0 +1,113 @@
+        worksheet = read_excel(file_path, sheet_name='Usage Reporting')
+        cell_value = worksheet.iat[3, 1]
+
+        assert cell_value == '2025-02-25, 2025-02-26', f"Expected '2025-02-25, 2025-02-26' but got '{cell_value}'"
+    finally:
+        workbook.close()
+        os.remove(file_path)
+
+
+month_file_path = './metrics_utility/test/test_data/reports/2025/04/CCSPv2-2025-04.xlsx'
+
+
+def test_accepts_month_and_ignores_since_until():
+    env_vars['METRICS_UTILITY_REPORT_TYPE'] = 'CCSPv2'
+    args = {'since': '2024-01-01', 'until': '2024-02-01', 'month': '2025-04', 'force': True}
+    # Build report, then assert that the report contains only dates of the month provided
+    run_build_int(env_vars, args)
+    try:
+        workbook = load_workbook(filename=month_file_path)
+        worksheet = read_excel(month_file_path, sheet_name='Usage Reporting')
+        cell_value = worksheet.iat[3, 1]
+        assert cell_value == '2025-04', f"Expected '2025-04' but got '{cell_value}'"
+    finally:
+        workbook.close()
+        os.remove(month_file_path)
+
+
+def test_renewal_guidancev2_fails_with_until_params(caplog):
+    command_args = [
+        {'args': {'until': '2025-01-01'}, 'report_type': 'RENEWAL_GUIDANCE'},
+        {'args': {'until': '2025-01-01'}, 'report_type': 'RENEWAL_GUIDANCEv2'},
+        {'args': {'until': '2025-01-01', 'since': '2024-02-01'}, 'report_type': 'RENEWAL_GUIDANCE'},
+    ]
+    for arg in command_args:
+        handle_build_exception(
+            {**env_vars, 'METRICS_UTILITY_REPORT_TYPE': arg['report_type']},
+            arg['args'],
+            'The --until parameter is not allowed when environment variable METRICS_UTILITY_REPORT_TYPE is RENEWAL_GUIDANCE or RENEWAL_GUIDANCEv2',
+        )
+
+
+def test_invalid_month_format():
+    bad_inputs = ['abc', '12', 'mo3', '3mon', '3months', '3m', '3mo']
+
+    for bad_input in bad_inputs:
+        handle_build_exception(env_vars, {'month': bad_input}, 'Invalid --month format. Supported date format: YYYY-MM')
+
+
+def handle_build_exception(env_vars, params, error_message):
+    with pytest.raises(SystemExit) as exc_info:
+        run_build_int(env_vars, params)
+
+        # Optionally check exit code
+        assert exc_info.value.code == 1
+        found_error = any(error_message in record.message for record in exc_info.value)
+        assert found_error, f'Expected error message not found in logs: {exc_info.value}'
+
+
+gather_env_vars = {
+    'METRICS_UTILITY_REPORT_TYPE': 'CCSPv2',
+    'METRICS_UTILITY_SHIP_PATH': './metrics_utility/test/test_data',
+    'METRICS_UTILITY_SHIP_TARGET': 'directory',
+}
+
+
+def handle_gather_exception(env_vars, params, error_message):
+    with pytest.raises(Exception) as exc_info:
+        run_gather_int(env_vars, params)
+
+        found_error = any(error_message in record.message for record in exc_info.value)
+        assert found_error, f'Expected error message not found in logs: {exc_info.value}'
```

Comment:
SonarQube was right here, there are 0 asserts in this test...

`run_gather_int` raises an exception, as expected, ending that block
python then continues after the `with`  .. but there's no code left there

```suggestion
    with pytest.raises(Exception) as exc_info:
        run_gather_int(env_vars, params)

    found_error = any(error_message in record.message for record in exc_info.value)
    assert found_error, f'Expected error message not found in logs: {exc_info.value}'
```